packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -1784,20 +1786,35 @@ function warnOnOpaqueIdentifierAccessInDEV(fiber) {
     const name = getComponentName(fiber.type) || 'Unknown';
     if (getIsRendering() && !didWarnAboutUseOpaqueIdentifier[name]) {
       console.error(
-        'The object passed back from useOpaqueIdentifier is meant to be ' +
-          'passed through to attributes only. Do not read the ' +
-          'value directly.',
+        'Warning: The object passed back from useOpaqueIdentifier and those made from it are meant to be passed through to attributes only. Do not read the value directly.',
       );
       didWarnAboutUseOpaqueIdentifier[name] = true;
     }
   }
 }
 
+function warnOnDuplicateOpaqueIdentifierKeyInDEV(fiber) {
+  if (__DEV__) {
+    const name = getComponentName(fiber.type) || 'Unknown';
+    if (getIsRendering() && !didWarnAboutDuplicateOpaqueIdentifierKeys[name]) {
+      console.error(
+        'Warning: Detected that two nested opaque identifiers with the same parent were created with identical keys. Keys must be unique.',
```

Comment:
Should we add name in this debug/error?

Reply:
Name is just the fiber name. I just copied the similar error above it.